calcite-qml/Calcite/SpinBox.qml

```diff
@@ -0,0 +1,188 @@
+/*******************************************************************************
+ *  Copyright 2012-2022 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+import QtQuick 2.15
+import QtQuick.Templates 2.15 as T
+
+T.SpinBox {
+    id: control
+
+    editable : true
+    implicitWidth: Math.max(implicitBackgroundWidth + leftInset + rightInset,
+                            contentItem.implicitWidth +
+                            up.implicitIndicatorWidth +
+                            down.implicitIndicatorWidth)
+    implicitHeight: Math.max(implicitContentHeight + topPadding + bottomPadding,
+                             implicitBackgroundHeight,
+                             up.implicitIndicatorHeight,
+                             down.implicitIndicatorHeight)
+    topPadding: 5
+    bottomPadding: 5
+
+    spacing: 6
+//    topPadding: 16
+//    bottomPadding: 16
+    leftPadding: (control.mirrored ? (up.indicator ? up.indicator.width : 0) : (down.indicator ? down.indicator.width : 0))
+    rightPadding: (control.mirrored ? (down.indicator ? down.indicator.width : 0) : (up.indicator ? up.indicator.width : 0))
+
+    validator: IntValidator {
+        locale: control.locale.name
+        bottom: Math.min(control.from, control.to)
+        top: Math.max(control.from, control.to)
+    }
+
+    contentItem: TextInput {
+        id: textInput
+        text: control.displayText
+
+        font: control.font
+        //color: enabled ? Calcite.text3 : Calcite.text2
+        color : Calcite.text1
+        selectionColor: Calcite.brandHover
+        selectedTextColor: Calcite.foreground1
+        horizontalAlignment: Qt.AlignHCenter
+        verticalAlignment: Qt.AlignVCenter
+
+        //cursorDelegate: CursorDelegate { }
+
+        readOnly: false
+        //readOnly: !control.editable
+        validator: control.validator
+        inputMethodHints: control.inputMethodHints
+    }
+
+    up.indicator: Item {
+        x: control.mirrored ? 0 : parent.width - width
+        implicitWidth: 40
+        implicitHeight: 40
+        height: parent.height
+        width: height
+        Rectangle {
+            //y: parent.y + parent.height - height - control.bottomPadding / 2
+            width: parent.width
+            height: parent.width - bottomPadding * 2
+            y : bottomPadding
```

Comment:
🔧 By the looks of things, this only works if and when `topPadding == bottomPadding`, and will look off-center if the developer were to set different topPadding and bottomPadding independently on their SpinBox.